File: ntbs-service/Models/NotificationInfo.cs
using System.ComponentModel.DataAnnotations.Schema;
using ntbs_service.Models.Entities;

namespace ntbs_service.Models
{
    [NotMapped]
    public class NotificationInfo
    {
        public string Name { get; set; }
        public string Dob { get; set; }
        public string NhsNumber { get; set; }
        public string DrugResistance { get; set; }
        public string Sex { get; set; }
        public string Postcode { get; set; }

        public static NotificationInfo CreateFromNotification(Notification notification)
        {
            return new NotificationInfo
            {
                Name = notification.PatientDetails.FullName,
                Dob = notification.PatientDetails.FormattedDob,
                NhsNumber = notification.PatientDetails.FormattedNhsNumber,
                DrugResistance = notification.DrugResistanceProfile.DrugResistanceProfileString,
                Sex = notification.PatientDetails.SexLabel,
                Postcode = notification.PatientDetails.FormattedNoAbodeOrPostcodeString
            };
        }
    }
}

File: ntbs-service/wwwroot/source/Components/NotificationInfo.ts
import Vue from "vue";

const NotificationInfo = Vue.extend({
    props: ["notificationInfo"],
    template:`
        <div id="notification-info" class="notification-info">
            <p>Name: {{notificationInfo.Name}}</p>
            <p>DOB: {{notificationInfo.Dob}}</p>
            <p>NHS number: {{notificationInfo.NhsNumber}}</p>
            <p>Drug resistance profile: {{notificationInfo.DrugResistance}}</p>
            <p>Sex: {{notificationInfo.Sex}}</p>
            <p>Postcode: {{notificationInfo.Postcode}}</p>
        </div>
    `
});

export {
    NotificationInfo
};